evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
You may have to look in the diff for https://github.com/pipermerriam/py-evm/pull/12 to see the how `get_vm_class_for_block_number()` elsewhere as I believe those uses are only present in that branch.

Short answer is that the `db` property needs to be set as a class level property since it will be used in both normal methods and class methods.